gemrb/GemRB.cpp

```diff
@@ -52,17 +52,13 @@ int main(int argc, char* argv[])
 
 	Interface::SanityCheck(VERSION_GEMRB);
 
-	core = new Interface();
-
-	if (core->Init(LoadFromArgs(argc, argv)) == GEM_ERROR) {
-		delete core;
-		Log(MESSAGE, "Main", "Aborting due to fatal error...");
-
-		return -1;
+	try {
+		Interface gemrb(LoadFromArgs(argc, argv));
+		gemrb.Main();
+	} catch (std::exception& e) {
```

Comment:
Do we throw anything but runtime_error? If not, we can be more specific here and also avoid a sonar warning.

Reply:
yeah, I'll just typedef something like `InitializationException`. Technically we can throw other exceptions, eg if we call `.at` on a container with an out of bounds index, but we can just let those be uncaught and terminate that way, or do a second catch block for things that shouldn't happen.